rest-api/src/main/java/com/cloudbees/workflow/rest/external/StageNodeExt.java

```diff
@@ -55,23 +57,7 @@ public void setStageFlowNodes(List<AtomFlowNodeExt> stageFlowNodes) {
     }
 
     public static boolean isStageNode(FlowNode node) {
-        return (node.getAction(StageAction.class) != null);
-    }
-
-    public static StageNodeExt create(FlowNode node) {
-        StageNodeExt stageNodeExt = new StageNodeExt();
-
-        stageNodeExt.addBasicNodeData(node);
-
-        // Use the last node in the stage to configure the stage status.
-        FlowNode stageEndNode = FlowNodeUtil.getStageEndNode(node);
-        if (NotExecutedNodeAction.isExecuted(stageEndNode)) {
-            stageNodeExt.setStatus(StatusExt.valueOf(stageEndNode.getError()));
-        } else {
-            stageNodeExt.setStatus(StatusExt.NOT_EXECUTED);
-        }
-
-        return stageNodeExt;
+        return (node.getAction(LabelAction.class) != null && node.getAction(ThreadNameAction.class) == null);
```

Comment:
> if added in the future

This will already match some non-`stage` block nodes _today_, but I think that is what we want.

Is this code only called in the case of block nodes? Because it will certainly match plenty of non-block-scoped steps which add `LabelAction` for various reasons, which I think we do _not_ want to detect in this plugin—we only care about (a) legacy `stage` (which adds `LabelAction` and `StageAction`), and (b) block-scoped steps (including but not limited to new `stage`) which add `LabelAction` (but specifically excluding `parallel` which adds `ThreadNameAction`).